pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/internal/TestFrameworksUtil.java

```diff
@@ -79,6 +79,8 @@ public static boolean isTestConfigurationMethod(ASTMethodDeclaration method) {
                 || method.isAnnotationPresent("org.junit.BeforeClass")
                 || method.isAnnotationPresent("org.junit.After")
                 || method.isAnnotationPresent("org.junit.AfterClass")
+                || method.isAnnotationPresent("org.testng.annotations.AfterClass")
+                || method.isAnnotationPresent("org.testng.annotations.BeforeClass")
```

Comment:
nice catch! There are however a bunch of other TestNG annotations that are missing:

`@BeforeMethod`, `@AfterMethod`, `@BeforeGroup`, `@AfterGroup`, `@BeforeTest`, `@AfterTest`, `@BeforeSuite` and `@AfterSuite`…

To keep this method manageable, we probably want to move all these annotation name Strings (both for junit and testng) to a private static immutable `Set<String>`, and replace all these `method.isAnnotationPresent` with something along the lines:

```java
return TEST_CONFIGURATION_ANNOTATIONS.stream().anyMatch(method::isAnnotationPresent)
    || …
```

Reply:
Thanks for comment. I will revise it later.